internal/smtp/net.go

```diff
@@ -0,0 +1,57 @@
+//go:generate mockgen -typed -source=net.go -destination=net_mock.go -package=smtp
+package smtp
+
+import (
+	"fmt"
+	"kg/procurement/cmd/config"
+	"log"
+	gosmtp "net/smtp"
+	"strings"
+)
+
+type SMTPClient interface {
```

Comment:
Why is there a "Real" STMPClient?

Reply:
Because we unfortunately cannot mock the API that `net/smtp` provides. For instance, the `PlainAuth` would require us to provide an actual working email (exposing my own email sadly).

Hence, to ensure the correct flow, I decided to mock the actual interface, such that we could at least confirm the flow, however it would not reach 100% coverage, because of the reasons stated above.